esmvalcore/preprocessor/_units.py

```diff
@@ -29,3 +30,42 @@ def convert_units(cube, units):
     """
     cube.convert_units(units)
     return cube
+
+
+def flux_to_total(cube):
+    """Convert flux to aggregated values.
+
+    Flux should have units of X s-1 or any compatible unit.
+
+    Parameters
+    ----------
+    cube : iris.cube.Cube
+        Data cube for the flux
+
+    Returns
+    -------
+    iris.cube.Cube
+        Cube with the aggregated data
+
+    Raises
+    ------
+    ValueError
+        If the units are not supported by the operator
+    """
+    if 's-1' not in cube.units.format():
+        raise ValueError(
+            f'Units {cube.units} do not contain a supported flux definition.'
+            )
```

Comment:
This is no longer needed

Reply:
I am not sure if it makes sense to be able to multiply data with any unit by a factor that comes from any coordinate value (`percentage * degrees_east` for instance). 

The way I see this is that this is a preprocessor that allows to go from a variable representing a rate to a "fixed" quantity by multiplying the data by the unit in the denominator. That is why it works for the time, because the units of the time coordinate can match the units of variables that represent a rate. I don't think there are any variables using `degrees_east` or `degrees_north` as units.

But of course I am not aware of all possible applications.

Reply:
What is the problem you see with `% * degrees_east`? It may be a bit unusual, but I don't see anything wrong with it from a mathematical or physical point of view, hence I see no reason to forbid it; we can leave that decision up to the user.

Reply:
In the case of longitude, it would probably make more sense to multiply by the cell area, combining latitude and longitude. As I understand it now, this is the same preprocessor as the integration preprocessors, just without the final summation. Does that sound right?

Reply:
I wouldn't think of this as a special case. Rather, let's disallow multi-dimensional coordinates for now and possibly extend this later.

Reply:
@bouweandela does this last suggestion sound good to you?

Reply:
Sure, no need to implement everything at the same time

Reply:
However I can't think of a decent name for this

Reply:
@ESMValGroup/esmvaltool-developmentteam Anyone a suggestion for a good name for this preprocessor function? At the moment, it can multiply a variable by (1D) weights computed from the coordinate bounds. Right now only for the time coordinate, but this could trivially be extended to any coordinate.

In the future other weights might be possible (maybe cell area, volume, ..) if anyone has a use case and is interested in implementing that, though that could also go into a different preprocessor.

Reply:
I'll just name it `accumulate_coordinate`. If anyone strongly opposes to that let us know.